pkgs/development/libraries/gstreamer/bad/default.nix

```diff
@@ -319,8 +317,8 @@ stdenv.mkDerivation rec {
     "-Dnvcodec=disabled" # Linux-only
   ] ++ lib.optionals (!stdenv.hostPlatform.isLinux || !gst-plugins-base.waylandEnabled) [
     "-Dva=disabled" # see comment on `libva` in `buildInputs`
-  ] ++ lib.optionals (!stdenv.hostPlatform.isLinux || !guiSupport) [
-    "-Ddirectfb=disabled"
+  ] ++ [
+    (lib.mesonEnable "directfb" false)
```

Comment:
It would make sense to move this up to he non-conditional list instead of having this in the middle between conditionals. E.g. under `(lib.mesonEnable "doc" enableDocumentation)`.